bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java

```diff
@@ -154,9 +154,13 @@ public Connection(@Nullable Connection oldConnection, Gson gson) {
         String serial = null;
         String deviceId = null;
         if (oldConnection != null) {
-            frc = oldConnection.getFrc();
-            serial = oldConnection.getSerial();
             deviceId = oldConnection.getDeviceId();
+            if (checkDeviceIdIsValid(deviceId)) {
```

Comment:
I think, it would be better to do the check in the getLoginPage function. Otherwise it could happen that still working accounts gets logout out.